### PR TITLE
Catch up with v0.10 API/SPI

### PR DESCRIPTION
**Pull request by dmikurube**
Catching up with Embulk v0.10 API/SPI. See https://dev.embulk.org/topics/get-ready-for-v0.11-and-v1.0.html for the details.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,45 +1,89 @@
...
+    }
+
+    // They are once excluded from transitive dependencies of other dependencies,
+    // and added explicitly with versions exactly the same with embulk-core:0.10.31.
+    compile "com.fasterxml.jackson.core:jackson-annotations:2.6.7"
+    compile "com.fasterxml.jackson.core:jackson-core:2.6.7"
+    compile "com.fasterxml.jackson.core:jackson-databind:2.6.7"
+    compile "com.fasterxml.jackson.datatype:jackson-datatype-jdk8:2.6.7"
+    compile "javax.validation:validation-api:1.1.0.Final"
+    compile "joda-time:joda-time:2.9.2"
```
Comment (vietnguyen-td) [inline build.gradle:64]: still use joda for backward compatibility?

**Review by vietnguyen-td — COMMENTED** [review]
just one conern about joda

File: build.gradle (hunk 1 above)
Comment (dmikurube) [reply build.gradle:64]: This `joda-time` is used from `aws-java-sdk`, so we have to include this explicitly.

**Review by vietnguyen-td — APPROVED** [review]
LGTM

**Comment by dmikurube** [comment]
Thanks!

**Pull request merged**